velox/functions/prestosql/tests/MapFromEntriesTest.cpp

```diff
@@ -0,0 +1,161 @@
+ */
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+std::optional<std::vector<std::pair<int32_t, std::optional<int32_t>>>> O(
+    const std::vector<std::pair<int32_t, std::optional<int32_t>>>& vector) {
+  return std::make_optional(vector);
+}
+} // namespace
+
+namespace {
+class MapFromEntriesTest : public FunctionBaseTest {
+ protected:
+  void verifyMapFromEntriesWithTry(
+      const VectorPtr& input,
+      const VectorPtr& expected) {
+    verifyMapFromEntries(input, expected, true);
+  }
+
+  void verifyMapFromEntries(
+      const VectorPtr& input,
+      const VectorPtr& expected,
+      bool wrappedWithTry = false) {
+    const std::string expr =
+        wrappedWithTry ? "try(map_from_entries(C0))" : "map_from_entries(C0)";
+    auto result = evaluate<MapVector>(expr, makeRowVector({input}));
+    assertEqualVectors(expected, result);
+  }
+
+  // Evaluate an expression only, usually expect error thrown.
+  void evaluateExpr(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input) {
+    evaluate(expression, makeRowVector(input));
+  }
+
+  void evaluateExprWithConstantArray(
+      const std::vector<MapVectorPtr>& expectedVectors,
+      const std::string& expression,
+      const VectorPtr& input,
+      const vector_size_t kConstantSize = 1'000) {
+    auto evaluateConstant = [&](vector_size_t row, const VectorPtr& vector) {
+      return evaluate(
+          expression,
+          makeRowVector(
+              {BaseVector::wrapInConstant(kConstantSize, row, vector)}));
+    };
+    for (auto row = 0; row < expectedVectors.size(); row++) {
+      auto result = evaluateConstant(row, input);
+      auto expectedConstant =
+          BaseVector::wrapInConstant(kConstantSize, 0, expectedVectors[row]);
+      assertEqualVectors(expectedConstant, result);
+    }
+  }
+};
+} // namespace
+
+TEST_F(MapFromEntriesTest, intKeyAndVarcharValue) {
+  auto rowType = ROW({INTEGER(), VARCHAR()});
+  std::vector<std::vector<variant>> data = {
+      {variant::row({1, "red"}),
+       variant::row({2, "blue"}),
+       variant::row({3, "green"})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  auto expected = makeMapVector<int32_t, StringView>(
+      {{{1, "red"_sv}, {2, "blue"_sv}, {3, "green"_sv}}});
+  verifyMapFromEntries(input, expected);
+}
+
+TEST_F(MapFromEntriesTest, nullMapEntries) {
+  auto rowType = ROW({INTEGER(), INTEGER()});
+  std::vector<std::vector<variant>> data = {
+      {variant(TypeKind::ROW)}, {variant::row({1, 11})}};
+  auto input = makeArrayOfRowVector(rowType, data);
+  VELOX_ASSERT_THROW(
+      evaluateExpr("map_from_entries(C0)", {input}),
+      "map entry cannot be null");
+  auto expected =
+      makeNullableMapVector<int32_t, int32_t>({std::nullopt, O({{1, 11}})});
```

Comment:
Does it work to replace `O()` with an additional pair of `{}` and avoid the use of O() for readability?

Reply:
I don't think we can initialize a std::optional object by calling something like `{n}`, which is an initializer list, not a single value. 
There are also some existing code pointer I borrowed from, e.g. [this](https://github.com/facebookincubator/velox/blob/main/velox/vector/tests/VectorMakerTest.cpp#L347-L347).

Reply:
You can try it out, because I know in some tests we use `{n}` to initialize std::optional. If it doesn't work here, you can keep O().